src/facade/dragonfly_connection.cc

```diff
@@ -591,10 +593,15 @@ void Connection::ConnectionFlow(FiberSocketBase* peer) {
   // After the client disconnected.
   cc_->conn_closing = true;  // Signal dispatch to close.
   evc_.notify();
+  phase_ = SHUTTING_DOWN;
 
   VLOG(1) << "Before dispatch_fb.join()";
   dispatch_fb_.JoinIfNeeded();
   VLOG(1) << "After dispatch_fb.join()";
+
+  phase_ = PRECLOSE;
+
+  ClearPipelinedMessages();
```

Comment:
if we migrate a connection and join on the dispatching fiber, then now we do not clear pipelined messages  but before this change we cleared them because ` ClearPipelinedMessages` logic resided in the dispatch fiber.
Am I correct about this?

Reply:
Currently we only support migration with an empty dispatch queue, it's the logic of MigrationRequest message and we have dchecks for it all over the place

Reply:
if it's trigerred during migration which is rare, maybe worth converting some of them to CHECKs? CHECK-fails's are better than deadocks or UB.

Reply:
done